pkg/pipeline/templates/terraformed.go.tmpl

```diff
@@ -92,3 +92,25 @@ func (tr *{{ .CRD.Kind }}) LateInitialize(attrs []byte) (bool, error) {
 func (tr *{{ .CRD.Kind }}) GetTerraformSchemaVersion() int {
     return {{ .Terraform.SchemaVersion }}
 }
+
+{{- if .ExternalTagsFieldName }}
+// SetExternalTags for this {{ .CRD.Kind }}
+func (tr *{{ .CRD.Kind }}) SetExternalTags() {
+	if tr.Spec.ForProvider.{{ .ExternalTagsFieldName }} == nil{
+		tr.Spec.ForProvider.{{ .ExternalTagsFieldName }} = map[string]*string{}
```

Comment:
I suspect that the type won't always be `map[string]*string{}`, which if changed would affect the whole function.

What are your thoughts about having the function implemented in provider repositories and refer to it as initializer function in the template? Provider authors can use whatever logic they'd like to and also possibly add more initializers if they need.

Reply:
Hi @muvaf thanks for comment. Yes you are right that these fields are not always of a `map[string]*string{}` type. There are a few statistics about this in the description section of the PR. According to these data, it is very rare situation. However, it is possible that this situation may be seen more frequently in other providers. Therefore, it may be reasonable for us to handle this.

For the second paragraph of your comment:

https://github.com/crossplane-contrib/provider-jet-azure/blob/e04776f600c35b1c26e389937593b90fc6fbded1/config/kubernetes/config.go#L58

As you know, we have GetExternalNameFn field that is a function type. This field was set in providers side (by using the functions that are implemented in provider side). Do you mean something like this when the relevant function is implemented in providers?

Reply:
> According to these data, it is very rare situation.

Thanks for running these stats! I think accepting initializer will be useful for other stuff as well, so we can have the implementation you got here as the default, similar to other common defaults [here](https://github.com/crossplane/terrajet/blob/main/pkg/config/defaults.go#L28).

> Do you mean something like this when the relevant function is implemented in providers?

Yes, we can possibly accept an [`InitializerChain`](https://github.com/crossplane/crossplane-runtime/blob/295de47/pkg/reconciler/managed/reconciler.go#L143) and `DefaultResourceFn` can set it to the default set.

Reply:
I updated the PR according to your comments. The implementations of setting tag fields are moved to provider repositories. Also, with last changes, this PR transformed from a tagging initializer PR to a generic initializer PR.

I am putting the provider PR links here to show the all process:

https://github.com/crossplane-contrib/provider-jet-azure/pull/130
https://github.com/crossplane-contrib/provider-jet-aws/pull/151
https://github.com/crossplane-contrib/provider-jet-gcp/pull/37

For `Initializers` field in resource configuration struct, `InitializerChain` type was not used directly to pass the `client.Client` in controllers. Instead of this, an `InitializerFn` type was used. This function type gets a `client.Client` as a parameter and returns an `managed.Initializer`. In controllers, before setting the `ReconcilerOptions`, the `Initializers` (configuration field) items is iterated and the client passes to the all `managed.Initializer`s.